service/worker/service.go

```diff
@@ -283,6 +283,7 @@ func (s *Service) startESAnalyzer() {
 		s.GetFrontendClient(),
 		s.GetClientBean(),
 		s.params.ESClient,
+		s.params.PinotClient,
```

Comment:
nit: it's fine for this PR but as a general pattern we should avoid passing multiple client objects to a component and having it decide based on which one is nil. Ideally there would be one parameter here of `visibilityClient` type or something like that and the bootstrap code (this file) determines which one to pass there.